sdk/service/service.go

```diff
@@ -0,0 +1,89 @@
+package servicesdk
+
+import (
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/cskr/pubsub"
+	"github.com/docker/docker/pkg/archive"
+	"github.com/mesg-foundation/core/container"
+	"github.com/mesg-foundation/core/database"
+	"github.com/mesg-foundation/core/service"
+	"github.com/mesg-foundation/core/service/manager"
+	"github.com/mesg-foundation/core/utils/dirhash"
+	"github.com/mr-tron/base58"
+	uuid "github.com/satori/go.uuid"
+)
+
+// ServiceSDK exposes service APIs of MESG.
+type ServiceSDK struct {
+	ps *pubsub.PubSub
+
+	m         manager.Manager
+	container container.Container
+	db        database.ServiceDB
+	execDB    database.ExecutionDB
+}
+
+// New creates a new ServiceSDK with given options.
+func New(m manager.Manager, c container.Container, db database.ServiceDB, execDB database.ExecutionDB) *ServiceSDK {
+	return &ServiceSDK{
+		ps:        pubsub.New(0),
+		m:         m,
+		container: c,
+		db:        db,
+		execDB:    execDB,
+	}
+}
+
+// Create creates a new service from definition.
+func (s *ServiceSDK) Create(srv *service.Service) error {
+	// download and untar service context into path.
+	path, err := ioutil.TempDir("", "mesg-"+uuid.NewV4().String())
```

Comment:
`ioutil.TempDir` already create an unique folder. `uuid.NewV4().String()` is not necessary.

Reply:
done

Reply:
done